src/librustc/middle/free_region.rs

```diff
@@ -162,23 +162,23 @@ impl<'tcx> FreeRegionMap<'tcx> {
     /// (with the exception that `'static: 'x` is not notable)
     pub fn relate_regions(&mut self, sub: Region<'tcx>, sup: Region<'tcx>) {
         debug!("relate_regions(sub={:?}, sup={:?})", sub, sup);
-        if (is_free(sub) || *sub == ty::ReStatic) && is_free(sup) {
+        if is_free_or_static(sub) && is_free(sup) {
             self.relation.add(sub, sup)
         }
     }
 
-    /// True if `r_a <= r_b` is known to hold. Both `r_a` and `r_b`
-    /// must be free regions from the function header.
+    /// Tests whether `r_a <= sup`. Both must be free regions or
```

Comment:
... did you mean to alpha-rename `r_b` to `sup` in the signature below? Why change the name in the comment?